draft-ietf-dnssd-privreq.xml

```diff
@@ -817,7 +824,128 @@ This draft does not require any IANA action.
   </front>
 </reference>
 
+<reference anchor="Wu16" target="https://arxiv.org/pdf/1604.06959.pdf%22">
+  <front>
+    <title>Privacy, discovery, and authentication for the internet of things</title>
+    <author initials='D.' surname='Wu' fullname='David Wu'>
+      <organization>Stanford University</organization>
+    </author>
+    <author initials='A.' surname='Taly' fullname='Ankur Taly'>
+      <organization>Google</organization>
+    </author>
+    <author initials='A.' surname='Shankar' fullname='Asim Shankar'>
+      <organization>Google</organization>
+    </author>
+    <author initials='D.' surname='Boneh' fullname='Dan Boneh'>
+      <organization>Stanford University</organization>
+    </author>
+    <date year="2016"/>
+  </front>
+</reference>
+
+<reference anchor="SIGMA" target="http://link.springer.com/content/pdf/10.1007/978-3-540-45146-4_24.pdf">
+  <front>
+    <title>SIGMA: The 'SIGn-and-MAc'approach to authenticated Diffie-Hellman and its use in the IKE protocols</title>
+    <author initials='H.' surname='Krawczyk' fullname='Hugo Krawczyk'>
+      <organization>EE Department, Technion, Haifa, Israel, and IBM T.J. Watson Research Center</organization>
+    </author>
+    <date year="2003"/>
+  </front>
+</reference>
+
 </references>  
 
+  <section title="Survey of Implementations">
+
+    <t>This section surveys several private service discovery designs in the context of the threat model
+      detailed above.</t>
+
+    <section title="DNS-SD Privacy Extensions">
+      <t>Huitema and Kaiser <xref target="I-D.huitema-dnssd-privacy"></xref> decompose 
+      private service discovery into two stages: (1) identify specific peers offering 
+      private services, and (2) issue unicast DNS-SD queries to those hosts after 
+      connecting over TLS using a previously agreed upon pre-shared key (PSK), or 
+      pairing key. Any out-of-band pairing mechanism will suffice for PSK establishment, 
+      though the authors specifically mention <xref target="I-D.ietf-dnssd-pairing"></xref> 
+      as the pairing mechanism. Step (1) is done by broadcasting "private instance names"
+      to local peers, using service-specific pairing keys. A private instance name N' 
+      for some service with name N is composed of a unique nonce r and commitment to r using 
+      N_k. Commitments are constructed by hashing N_k with the nonce. Only owners of N_k 
+      may verify its correctness and, upon doing so, answer as needed. The draft recommends 
+      randomizing hostnames in SRV responses to minimize likability to specific hosts. 
+      Note that this does not prevent fingerprinting and tracking using that hostname.</t>
```

Comment:
Actually, both the 2-phase architecture and the randomizing or host names are meant to prevent fingerprinting and tracking:

1) Randomizing means using a different host name at each different venue, 

2) The two phase process hides which services are being discovered, and thus prevents using the set of services that a host looks at as a fingerprint.

Clearly, host name randomization is only effective if combined with MAC address randomization, otherwise the host can be tracked by its MAC address. It may be that the draft is not clear enough, but the idea is to publish a new host name each time the MAC address changes.

As for fingerprinting, we have to consider fingerprinting during discovery and fingerprinting when using the discovered services. IMHO, a discovery standard should just address the leaks arising during discovery, and I think that the current draft does that. There are all kinds of other leaks that should be addressed during usage, such as leaking the SNI in clear text, revealing the server's public key during a connection attempt or a connection replay, or using an application-specific traffic pattern. But these belong in a separate effort, not in discovery privacy.

Reply:
I'll update the text to clarify.

Reply:
Done.